hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -361,7 +361,7 @@ def unlink(self):
         analytic_timesheet_toremove = self.env['account.analytic.line']
         for sheet in self:
             analytic_timesheet_toremove += \
-                sheet.timesheet_ids.filtered(lambda t: not t.task_id)
+                sheet.timesheet_ids.filtered(lambda t: t.name == '/')
```

Comment:
Isn't a more determinant condition?

Reply:
Unfortunately no, it's used all across the file

Reply:
I think we need to set a specific sheet_id or something for avoiding ghost removals. Although of course your code is better than now.

Reply:
`sheet.timesheet_ids` is already linked by `sheet_id`

Reply:
Then I don't get it... is this for some kind of cleanup?

Reply:
Yes, when AALs are created from matrix, they make no sense w/o sheet, yet manually created ones should remain

Reply:
OK, then if this condition is harder enough, good to go.

Reply:
Yeah, since I've just erased couple of manual entries because of this :)